persistor.rb

```diff
@@ -0,0 +1,83 @@
+require 'json'
+
+class Persistor
+  def hydrate(classroom)
+    books = parse_books
+    people = parse_people(classroom)
+    rentals = parse_rentals(people, books)
+
+    {
+      'people' => people,
+      'books' => books,
+      'rentals' => rentals
+    }
+  end
+
+  def parse_books
+    file = 'books.json'
+
+    if File.exist? file
+      JSON.parse(File.read(file), create_additions: true)
+    else
+      []
+    end
+  end
+
+  def parse_rentals(people, books)
+    file = 'rentals.json'
+
+    if File.exist? file
+      JSON.parse(File.read(file)).map do |rental_json|
+        book = books.find { |current_book| current_book.title == rental_json['book_title'] }
+        person = people.find { |current_person| current_person.id == rental_json['person_id'].to_i }
```

Comment:
- [ ] on first start the application is firing and working as expected when closing the application the JSON files are created as well, however, when restarting the application is not firing and the user gets below error:
![Screenshot 2022-01-13 120202](https://user-images.githubusercontent.com/39240820/149318901-f04fe80f-9b88-49a4-af59-2ab634f3171a.png)

the reason the user gets this error is because the information in the JSON files is saved as an array of strings and it can't find the title in the string. in order to make it work this way the data in your JSON file should be saved as an array of objects which will look something like in the below example:
```
[{"Title": "hp", "Author": "jk"},{"Title": "hp2", "Author": "jk"}]
```

In the example I'm using the books array but this also applies for the people and rental array.

Reply:
hello, Michael, 
can you explain more what do you mean by, when restarting the application is not firing and the user gets below error: 
because it's working fine on my machine

Reply:
now i see the problem now but I fixed it someway else 
thanks for the hint